src/primefactorization.jl

```diff
@@ -384,3 +384,32 @@ function sumlist!(list::Vector{<:PrimeFactorization}, ind = 1:length(list))
     end
     return MPZ.mul!(s, _convert!(i, g))
 end
+
+#=
+# A cached binomial implementation.
+bcache = LRU{Tuple{BigInt, BigInt}, PrimeFactorization}(; maxsize=10^6)
+function primebinomial(n::BigInt, k::BigInt)
+    T = PrimeFactorization{eltype(eltype(factorialtable))}
+    if k == 0
+        return one(T)
+    end # guard
+    if haskey(bcache, (n, k))
+        return bcache[(n, k)]
+    else
+        den = primefactor(k)
+        num = mul!(copy(primefactor(n + 1 - k)), primebinomial(n, k - 1)) 
+        res = divexact!(num, den)
+        bcache[(n, k)] = res
+        return res
+    end
+end
+=#
+
+function primebinomial(n::BigInt, k::BigInt)
+    num = copy(primefactorial(n))
+    den = copy(primefactorial(k))
+    den = mul!(den, primefactorial(n - k))
```

Comment:
if you simply do
```julia
divexact!(num, primefactorial(k))
divexact!(num, primefactorial(n-k))
```
there is no need to take a copy of `primefactorial(k)`. `divexact!` should be as quick as `mul!`, as it is just subtracting rather than adding prime powers, so there is no need to avoid doing it twice by first doing a `mul!`

Reply:
Thanks! Duh, will improve that!